Common/src/CConfig.cpp

```diff
@@ -4657,6 +4664,52 @@ void CConfig::SetPostprocessing(unsigned short val_software, unsigned short val_
     }
   }
 
+  /*--- Check that if the wall roughness array are compatible and set deafult values if needed. ---*/
+
+  if ((nMarker_HeatFlux > 0) || (nMarker_Isothermal > 0)) {
+    /*--- If the config option is not declared, then assume smooth walls and set roughness to zero. ---*/
+    if (nWall_Types != nMarker_HeatFlux + nMarker_Isothermal) {
+      /*--- If this option is not used at all, assume all walls are smooth and set roughness height to zero. ---*/
+      if (nWall_Types == 0) {
+        nWall_Types = nMarker_HeatFlux + nMarker_Isothermal;
+        Roughness_Height = new su2double [nWall_Types];
+        Kind_Wall = new unsigned short [nWall_Types];
+        for (unsigned short iMarker = 0; iMarker < nMarker_HeatFlux; iMarker++) {
+          Roughness_Height[iMarker] = 0.0;
+          Kind_Wall[iMarker] = SMOOTH;
+        }
+        for (iMarker = 0; iMarker < nMarker_Isothermal; iMarker++) {
+          Roughness_Height[nMarker_HeatFlux + iMarker] = 0.0;
+          Kind_Wall[nMarker_HeatFlux + iMarker] = SMOOTH;
+        }
+      }
+      else
+        SU2_MPI::Error("Mismatch in number of wall markers. Specify type of wall for all markers (even if smooth).", CURRENT_FUNCTION);
+    } else {
+       if (nRoughWall != nWall_Types)
+         SU2_MPI::Error("Mismatch in number of roughness height definition and wall type definition.", CURRENT_FUNCTION);
+       else {
+         vector<su2double> temp_rough;
+         vector<unsigned short> temp_kindrough;
+         for (unsigned short iMarker = 0; iMarker < nWall_Types; iMarker++) {
+           temp_rough.push_back(Roughness_Height[iMarker]);
+           temp_kindrough.push_back(Kind_Wall[iMarker]);
+         }
+         for (iMarker = 0; iMarker < nMarker_HeatFlux; iMarker++) {
+           Roughness_Height[iMarker] = temp_rough[iMarker];
+           Kind_Wall[iMarker] = temp_kindrough[iMarker];
+         }
+         for (iMarker = 0; iMarker < nMarker_Isothermal; iMarker++) {
+           Roughness_Height[nMarker_HeatFlux + iMarker] = temp_rough[nMarker_HeatFlux + iMarker];
+           Kind_Wall[nMarker_HeatFlux + iMarker] = temp_kindrough[nMarker_HeatFlux + iMarker];
+         }
+         /*--- Release memory allocated earlier. ---*/
+         vector<su2double> ().swap(temp_rough);
+         vector<unsigned short> ().swap(temp_kindrough);
```

Comment:
Based on advice from @pcarruscag I have used std::vector here. Apparently the clear function does not release the memory, so I used swap() instead (https://stackoverflow.com/questions/10464992/c-delete-vector-objects-free-memory). This is also used elsewhere in the code.

Reply:
Please remove the swaps, objects are destroyed when they go out-of-scope, the use cases for explicitly deleting vectors this way is when they are class members, in which case their scope is the same of the object that owns them.